Module 3, Block 5 : React capstone project - Metrics webapp

aceDavon opened the pull request:
## In This PR, I have implemented the following tasks: :writing_hand: 

- [ ] Use React documentation.
- [ ] Use React components.
- [ ] Use React props.
- [ ] Use React Router.
- [ ] Connect React and Redux.
- [ ] Handle events in a React app.
- [ ] Write integration tests with a React testing library.
- [ ] Use styles in a React app.
- [ ] Use React life cycle methods.
- [ ] Apply React best practices and language style guides in code.
- [ ] Use store, actions and reducers in React.

aceDavon committed:
Add reducer slice to store

Create and add the category and search slices to the store for making
API calls

aceDavon committed:
Extract and implement app features

Extract the category component to a stand alone component, Implement the
filter by category feature and conditionally render the UI from the
filtered results

aceDavon committed:
Disable nested tenary expression check

Disable linters check for nested tenary expressions to allow for
conditional rendering of the component UI using nested tenary expressions

aceDavon committed:
Extract and implement Search

Extract the search component and implement search from the API on click
of the search button

aceDavon committed:
Create Project documentation

Craete project documentation and write unit tests for both component and slice functions

aceDavon committed:
Fix linter errors on Test functions

aceDavon committed:
Configure project for deployment
 Configure project to check broken netlify routing

aceDavon committed:
Add deployment URL to documentation

Deploy project on Netlify and add live project URL to README.md

aceDavon committed:
Add video persentation Link

Record, upload and add video persentation link to README.md

topeogunleye commented on src/App.js:
```diff
@@ -13,6 +14,7 @@ function App() {
   return (
     <>
       <Header />
```
## As required, I kindly suggest that you follow the design guidelines in terms of: 
- [ ] Typography: font face, size, and weight i.e font used are lato and gill sans
- [ ]  Layout: composition and space between elements. Please reduce the spacing between the items on your homepage to match the DESIGN given:

| Your Design | Figma Design |
| ------------- | ------------- |
| <img width="812" alt="Screenshot 2022-10-05 192254" src="https://user-images.githubusercontent.com/52934128/194146644-6fbafcf0-44bf-4236-9516-9d5c3899e7e2.png">  | <img width="166" alt="image" src="https://user-images.githubusercontent.com/52934128/194146845-58a87796-6d1c-440b-bfa1-9abf7af027fc.png">  |

## You can select one main colour for the app and be consistent (it does **not** need to be the same colour as in the design guidelines)

aceDavon replied:
@topeogunleye  Thanks for calling my attention to this issue, I will fix this immediately.

topeogunleye commented on src/redux/search/searchIndex.js:
```diff
@@ -0,0 +1,90 @@
+import Spinners from '../../components/spinners';
+import { selectAllJobs } from './searchSlice';
+
+const SearchIndex = () => {
```
- [ ] On your Home page, your filter is supposed to filter through the items already on the homepage and not navigate to the category directly?


https://user-images.githubusercontent.com/52934128/194131352-4920c6e2-faa9-4414-80ca-c090ec904f18.mp4

aceDavon replied:
Thanks again @topeogunleye for the review, I will restructure my codebase to bear these changes as required herein. This wasn't possible with the present implementation but, I already got a cue on how to make this work. Thanks again for your review.

topeogunleye replied:
you're welcome

topeogunleye commented on src/redux/category/categoryIndex.js:
```diff
@@ -0,0 +1,54 @@
+import { Search, selectAllJobs } from '../search/searchSlice';
+import Spinners from '../../components/spinners';
+
+const CategoryIndex = () => {
```
- [ ] As required, I kindly suggest that you add a "Back" button on your details page so that it's possible to navigate back to the home page from the details page.

| Your Design  | Figma Design |
| ------------- | ------------- |
| <img width="946" alt="image" src="https://user-images.githubusercontent.com/52934128/194122696-f437d263-1571-4bec-a081-f34fa7aa5ac4.png">  | <img width="200" alt="image" src="https://user-images.githubusercontent.com/52934128/194127604-024f9180-7e1c-4b83-8f7e-23e822f695cf.png">  |

aceDavon replied:
Noted and Implemented!

topeogunleye commented on src/welcome.js:
```diff
@@ -20,12 +20,12 @@ const Welcome = () => {
             const { tag, label } = x;
```
- [ ] I kindly suggest that you organize your folders better. Your `categoryIndex.js` and `searchIndex.js` are not to be inside your redux folder, along with your test folders and files. they supposed to be in your components folder. Pls see below for reference on how to organize your files and folders:

```
app-root-directory
  |- package.json
  |- README.md
  |- /public
    |- index.html
  |- /src
    |- app.css
    |- index.js
    |- /components
    |- /container
    |- /pages
    |- /redux
  |- /node_modules
```

aceDavon replied:
Noted and resolved!

topeogunleye reviewed (changes requested):
Hi @aceDavon, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
## (Highlights) Good Points: 
* Good commit messages. :heavy_check_mark:
* No linter errors. :heavy_check_mark:
* Followed Github flow :heavy_check_mark:
* Professional README file. :heavy_check_mark:

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

aceDavon committed:
Refactor code to search store data

Refactor code to implement search on store data fetch on the UI

aceDavon committed:
Change site assets and logo

codepantha commented on src/welcome.js:
```diff
@@ -15,22 +15,80 @@ const Welcome = () => {
+                    <span className="text-xs font-thin text-slate-400">
+                      {category.label}
+                    </span>
+                  </div>
```
- [ ] [OPTIONAL] It would be nice if you made some cards a darker shade following a specific pattern just like the design guidelines 👍 
### Design Guideline
![image](https://user-images.githubusercontent.com/46839548/194316409-d57f3fc4-2bca-4348-b1a8-016335eb917f.png)

### Your Design
![image](https://user-images.githubusercontent.com/46839548/194316242-f3dc44ac-69ec-4f7a-a461-b6a6935dc9ae.png)

aceDavon replied:
Noted and Implemented

codepantha reviewed (changes requested):
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @aceDavon 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

<img src="https://media2.giphy.com/media/xUOxfh6ZM75efM3Bqo/giphy-downsized-medium.gif"/>

## Highlights 🎯 
✔️ Good commit messages
✔️ Used GitHub Flow
✔️ Professional Readme
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

- [x] As suggested by the previous reviewer, I suggest that you organize your folders better. Your `categoryIndex.js` and `searchIndex.js` are not to be inside your redux folder, along with your test folders and files. they are supposed to be in your components folder. Pls see below for reference on how to organize your files and folders:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

aceDavon committed:
Refactor folders structure

Extract components to components folder to implement request changes from
 review

Nemwel-Boniface reviewed (approved):
## Hello @aceDavon          :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 

![](https://media.giphy.com/media/3kuSo744UIPJjcJUEn/giphy.gif)

## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
